New reduced Web SDK documentation (#1636)
SoftwareAG/c8y-docs: content/web/gettingstarted.md

  - app_enablement
---

You can start your custom development by creating a default Angular application. You need to ensure that you scaffold the right
Angular version. The following table shows an overview of the supported versions:

| Angular version | Web SDK version | Comment |
| --- | --- | --- |
| 16.x.x | 19.x.x | |
| 15.x.x | 18.1.x - 18.x.x | Using `c8ycli` tooling, only yearly release |
| 14.x.x | 16.x.x - 18.0.x | Using `c8ycli` tooling |

{{< c8y-admon-info >}}
If you consider to use an older version then `19.x.x` you need to use our old tooling based on the `c8ycli` tool-set. Please [references to this documentation](ADDDOCLINKHERE!!!) in this case.
{{< /c8y-admon-info >}}  

## Scaffolding a new application for development

First, install the `@angular/cli` in the right version.

```bash
npx @angular/cli@v16-lts new --style=less
```

Second, change into the folder and add the `@c8y/websdk` package to your Angular application:

```bash
ng add @c8y/websdk
```

{{< c8y-admon-info >}}  

> Required is a node.js installation. If you have the wrong node.js version installed, the `npm install` step will prompt you with the needed version number.

Author: does not prompt you. It will inform you about the version number needed.

{{< /c8y-admon-info >}}

The CLI will prompt you in 2 simple steps for the version and the base template. Afterwards your application is set and you can start your first development server in step 3.

### Step 1: Select the version

```bash
? Which base version do you want to scaffold from? (Use arrow keys)
> 1019.0.X 
> 1019.X.0 
> other
```

> In the second step, the base scaffolding version must be selected. The interface will provide the

Author: didn't moved, this is actually the descriptiion for step 1.

last the latest available release. Additionally a version can be manually entered by selecting the
`other` option. If you do not know which version to select, we recommend to use the latest.

### Step 2: Select the base project to start from

```bash
? Which base project do you want to scaffold from?
  administration
  application
  cockpit
  devicemanagement
  hybrid
  tutorial
  widget-plugin
  package-blueprint
```

In step two, the base project to scaffold from must be selected. You can select any of the default
Cumulocity applications to reuse the functions provided there. In alternative, you could start a
blank application by selecting _"application"_ project. A complete list of all applications provided
can be found [here](#/getting-started/library/overview#c8yapps-extendable-applications)

### Step 3: Start the local development server

Now you are ready to start the application by running the `npm start` command. By default, the
application will proxy to the Cumulocity IoT cloud platform, however, you can proxy to a different
application using the `-u` flag. For example:

```bash
npm start -- -u http://mytenant.acme.iot
```

> When started, the application begins to compile. After it is compiled, you can navigate to

Author: the application ;)

Updated.

{{< /c8y-admon-info >}}

The CLI will prompt you in two steps for the version and the base template. Afterwards your application is set and you can start your first development server in step 3.

### Step 1: Select the version

```bash
? Which base version do you want to scaffold from? (Use arrow keys)
> 1019.0.X 
> 1019.X.0 
> other
```

In the first step, the base scaffolding version must be selected. The interface will provide the
last the latest available release. Additionally a version can be manually entered by selecting the
`other` option. If you do not know which version to select, we recommend to use the latest.

### Step 2: Select the base project to start from

```bash
? Which base project do you want to scaffold from?
  administration
  application
  cockpit
  devicemanagement
  hybrid
  tutorial
  widget-plugin
  package-blueprint
```

In step two, the base project to scaffold from must be selected. You can select any of the default
Cumulocity applications to reuse the functions provided there. In alternative, you could start a

> blank application by selecting _"application"_ project.

Reviewer: ```suggestion
blank application by selecting the "application" project.
```

Now you can start the application by running the `npm start` command. By default, the
application will proxy to the {{< product-c8y-iot >}} cloud platform, however, you can proxy to a different
application using the `-u` flag. For example:

```bash
npm start -- -u http://mytenant.acme.iot
```

When you start the command the application begins to compile. After it is compiled, you can navigate to
`http://localhost:4200/apps/<<your-app-name>>/` and login to your tenant.

{{< c8y-admon-info >}}  
You must provide your tenant name or the tenant ID on login (as the application cannot derive it from the URL on localhost). If you don't know your tenant name or the tenant ID you can click on your username in your tenant and get the information from the section Platform Information.
{{< /c8y-admon-info >}}  

You are now setup. Any changes you make to your local files will lead to recompiling. After a
refresh you will see your changes. 

### Create your first custom component

After creating the empty bootstrapping application you might want to start with your first content.
To do so, add a new component in the `src/app` to your project and save it as `hello.component.ts`:

> ```javascript

Reviewer 2: #trivial
```suggestion
```typescript
```

  - app_enablement
---

This guide will setup your first application. First step is to install the `@angular/cli` in the right version.

```bash
npx @angular/cli@v16-lts new --style=less
```

> Second, change into the folder and add the `@c8y/websdk` package to your Angular application:

Author: The command is called `cd` -> change dir ;) So I am assuming that people know that they are in a terminal context here.

> Second, change into the folder and add the `@c8y/websdk` package to your Angular application:

Reviewer: @janhommes alright. To keep it consistent I still propose to use "navigate to":

```suggestion
Second, navigate to the folder and add the `@c8y/websdk` package to your Angular application:
```
